src/Language/PureScript/Errors.hs

```diff
@@ -481,6 +483,8 @@ prettyPrintSingleError (PPEOptions codeColor full level showDocs relPath) e = fl
       line "The last statement in a 'do' block must be an expression, but this block ends with a let binding."
     renderSimpleErrorMessage OverlappingNamesInLet =
       line "The same name was used more than once in a let binding."
+    renderSimpleErrorMessage (TypeConstructorAlias _ _ _) =
+      line "The type synonym provided was an alias for a Type Constructor. These kinds of types can lead to confusing programs."
```

Comment:
Can we say this?

```
This type synonym aliases the type constructor XYZ. Consider using a newtype instead.
```

"confusing programs" is a bit vague, and we can give examples in the docs repo instead.